pkg/apis/pipeline/v1alpha1/run_validation.go

```diff
@@ -36,20 +36,30 @@ func (r *Run) Validate(ctx context.Context) *apis.FieldError {
 
 // Validate Run spec
 func (rs *RunSpec) Validate(ctx context.Context) *apis.FieldError {
+	// this covers the case rs.Ref == nil && rs.Spec == nil
 	if equality.Semantic.DeepEqual(rs, &RunSpec{}) {
 		return apis.ErrMissingField("spec")
 	}
 
-	if rs.Ref == nil {
-		return apis.ErrMissingField("spec.ref")
+	if rs.Ref != nil && rs.Spec != nil {
```

Comment:
need an additional check here, if both are `nil`

Reply:
Thanks, now I have understood ! Will open a PR with the fix soon.